src/lua.rs

```diff
@@ -71,6 +73,16 @@ impl Lua {
         create_lua(true)
     }
 
+    /// Creates a new Lua state from a state already declared. It is assumed `state` is valid.
+    pub(crate) unsafe fn from_state(state: *mut ffi::lua_State) -> Lua {
+        Lua {
+            state,
+            main_state: main_state(state),
+            ephemeral: true,
+            _phantom: PhantomData
+        }
+    }
```

Comment:
It's reasonable to want to make this a separate function, but creating an ephemeral Lua is only done in (now) two places. Either both should use this auxiliary function or neither should. Also it should be called something like make_ephemeral.

Reply:
The problem is that the module `hook` cannot access the fields of `Lua` and I didn't want to have `pub(crate)` everywhere. I will rename the function.